### PR TITLE
CLN: pandas/core/indexing.py

### PR DESCRIPTION
**Pull request by ivanovmg**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Slight clean-up of ``pandas/core/indexing.py``.

- Use ``suppress`` instead of ``try/except/pass`` (starting from Python 3.7)
- Slight rearrangement of statements
- Extract method ``_validate_key_length``

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -834,11 +831,9 @@ def _getitem_nested_tuple(self, tup: Tuple):
             if self.name != "loc":
                 # This should never be reached, but lets be explicit about it
                 raise ValueError("Too many indices")
-            try:
+            with suppress(IndexingError):
                 result = self._handle_lowerdim_multi_index_axis0(tup)
                 return result
```
Comment (phofl) [inline pandas/core/indexing.py:836]: You could avoid the declaration of result here.

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -782,14 +782,11 @@ def _getitem_lowerdim(self, tup: Tuple):
         # ...but iloc should handle the tuple as simple integer-location
         # instead of checking it as multiindex representation (GH 13797)
         if isinstance(ax0, ABCMultiIndex) and self.name != "iloc":
-            try:
+            with suppress(IndexingError):
                 result = self._handle_lowerdim_multi_index_axis0(tup)
                 return result
```
Comment (phofl) [inline pandas/core/indexing.py:787]: Same here

**Comment by jbrockmendel** [comment]
LGTM pending green.

If you're looking for high-value things to do in this file, narrowing what we're catching/suppressing is a never-ending battle.

File: pandas/core/indexing.py (hunk 2 above)
Comment (ivanovmg) [reply pandas/core/indexing.py]: Did that

File: pandas/core/indexing.py (hunk 1 above)
Comment (ivanovmg) [reply pandas/core/indexing.py]: Thank you! Changed.

**Comment by phofl** [comment]
Lgtm

**Comment by ivanovmg** [comment]
Can anyone re-start CI? Looks like there were internal CI engine failures.

**Comment by jreback** [comment]
thanks @ivanovmg nice cleanup

**Pull request merged**